### ISSUE TITLE
fix: 🤔 sd-menu – mouse click hinders keyboard behaviour

### ISSUE THREAD

**Issue by mariohamann** [solid-design-system/solid#2372]
## Current behavior

When I click on a menu button with the mouse, I can't interact with the keyboard afterwards.

https://github.com/user-attachments/assets/23ed69b7-9476-4ca1-8601-5016cc8cc3e2

## Expected behavior

It should have the same behaviour as when clicking with the enter button.

## Steps to reproduce

https://codepen.io/mariohamann/pen/OPNzWBK?editors=1110:
- Open Menu with Tab and Return: Keyboard works
- Open Menu with Mouse: Keyboard doesn't work

## DoR
- [ ] Item has business value
- [ ] Item has been estimated by the team
- [ ] Item is clear and well-defined
- [ ] Item dependencies have been identified

**Comment by yoezlem** [comment]
@martascorreia can we close the ticket or is something missing here?
